test/e2e/multiple_policies_for_same_node.go

```diff
@@ -0,0 +1,95 @@
+package e2e
+
+import (
+	"fmt"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	nmstatev1alpha1 "github.com/nmstate/kubernetes-nmstate/pkg/apis/nmstate/v1alpha1"
+)
+
+func infUpWithStaticIP(infName string, ipAddress string) nmstatev1alpha1.State {
+	return nmstatev1alpha1.NewState(fmt.Sprintf(`interfaces:
+    - name: %s
+      type: ethernet
+      state: up
+      ipv4:
+        address:
+        - ip: %s
+          prefix-length: 24
+        dhcp: false
+        enabled: true
+      ipv6:
+        enabled: false
+`, infName, ipAddress))
+}
+
+func infUpWithVlan(iface string, vlanId string) nmstatev1alpha1.State {
+	return nmstatev1alpha1.NewState(fmt.Sprintf(`interfaces:
+    - name: %s.%s
+      type: vlan
+      state: up
+      vlan:
+        base-iface: %s
+        id: %s
+`, iface, vlanId, iface, vlanId))
+}
+
+func infVlanAbsent(iface string, vlanId string) nmstatev1alpha1.State {
+	return nmstatev1alpha1.NewState(fmt.Sprintf(`interfaces:
+    - name: %s.%s
+      type: vlan
+      state: absent
+      vlan:
+        base-iface: %s
+        id: %s
+`, iface, vlanId, iface, vlanId))
+}
+
+func infStaticIPAbsent(iface string, ipAddress string) nmstatev1alpha1.State {
+	return nmstatev1alpha1.NewState(fmt.Sprintf(`interfaces:
+    - name: %s
+      type: ethernet
+      state: absent
+      ipv4:
+        address:
+        - ip: %s
+          prefix-length: 24
+        dhcp: false
+        enabled: true
+      ipv6:
+        enabled: false
+`, iface, ipAddress))
+}
+
+var _ = Describe("NodeNetworkState", func() {
+	Context("with multiple policies configured", func() {
+		var (
+			StaticIpPolicy = "static-ip-policy"
+			VlanPolicy     = "vlan-policy"
+			ipAddress      = string("10.244.0.1")
+			vlanId         = string("102")
+		)
+
+		BeforeEach(func() {
+			setDesiredStateWithPolicy(StaticIpPolicy, infUpWithStaticIP(*firstSecondaryNic, ipAddress))
+			setDesiredStateWithPolicy(VlanPolicy, infUpWithVlan(*firstSecondaryNic, vlanId))
+			waitForAvailablePolicy(StaticIpPolicy)
+			waitForAvailablePolicy(VlanPolicy)
+		})
+		AfterEach(func() {
+			setDesiredStateWithPolicy(StaticIpPolicy, infStaticIPAbsent(*firstSecondaryNic, ipAddress))
+			setDesiredStateWithPolicy(VlanPolicy, infVlanAbsent(*firstSecondaryNic, vlanId))
+			deletePolicy(StaticIpPolicy)
+			deletePolicy(VlanPolicy)
+		})
+		It("should have the IP and vlan interface configured", func() {
+			for _, node := range nodes {
+				ipAddressForNodeInterfaceEventually(node, *firstSecondaryNic).Should(Equal(ipAddress))
+				interfacesNameForNodeEventually(node).Should(ContainElement(fmt.Sprintf(`%s.%s`, *firstSecondaryNic, vlanId)))
+				vlanForNodeInterfaceEventually(node, fmt.Sprintf(`%s.%s`, *firstSecondaryNic, vlanId)).Should(Equal(vlanId))
+			}
+		})
+	})
+})
```

Comment:
is the rename
>  test/e2e/simple_bridge_and_bond.go → test/e2e/simple_bridge_and_bond_test.go 
related to this PR? Is it intentional (you do not add _test to your new file)

Reply:
I made 4 commites.
Only one is related to the test.
I'll split the PR....

Reply:
urghh, I prefer gerrit review process to github's every day. If you think that these commits are related you may bulk them together, but please explain this in the PR leading message.